File: Python/Task495.py

import math
def isprime(n):
  result = True
  max_div = round(math.sqrt(n))
  for p in range(2, max_div + 1):
    if (n % p == 0):
      result = False
      break
  return result


print('Введіть парне натуральне число для ')
print('перевірки гіпотези Гольдбаха >>')
m = int(input())
max_a = int(m / 2) + 1 # максимально можливе
# значення першого доданка шуканої суми
for a in range(2, max_a + 1):
  # якщо a складене, переходимо до 
  # наступного значення
  if not isprime(a):
    continue
  b = m - a
  if not isprime(b):
    continue
  # якщо  b складене, переходимо до 
  #  наступного значення a
  else:
    break
if (a == max_a):
  # якщо можливе значення першого доданка 
  # досягло максимуму, то потріний розклад
  # у вигляді суми не знайдено
  print('Гіпотеза Гольдбаха не підтверджується')
else:
  print(m, ' = ', a, ' + ', b)
  print('Гіпотезу Гольдбаха підтверджено')

File: Java/Task495.java
import java.util.Scanner;
public class Task495 {
  public static void main(String[] args) {
    Scanner scanner = new Scanner(System.in);
    System.out.println("Задайте парне натуральне " +
          "число для перевірки гіпотези Гольдбаха:\n");
    int m = scanner.nextInt();
    int max_a = m / 2 + 1;
    int a, b = 0;
    for (a = 2; a < max_a; a++)
    {
      // якщо a складене, переходимо до наступного значення
        // boolean-результат методу можемо використовувати
      // в якості умови для оператора розгалуження
      if (!isPrime(a)) continue;
       // тут вже відомо, що а просте число, бо, якщо a
       // не є простим оператор if вище перерве виконання
       // поточної ітерації і код нижче не виконуватиметься
      b = m - a;
       // якщо при цьому число b також виявиться простим
       // то цикл пошуку простих доданків можна завершити
       // достроково і перейти до повідомлення результату
      if(isPrime(b)) break;
    }
    if (a == max_a)
    {
      System.out.println("Гіпотезу Гольдбаха " + 
                                    "не підтверджено");
    }
    else
    {
      System.out.printf("%10d  = %10d + %10d\n", m, a, b);
      System.out.println("Гіпотезу Гольдбаха" + 
                                        " підтверджено");
    }
    scanner.close();
  }

    // для "внутрішнього" використання в методі main
    // метод-підпрограму можна оголошувати з будь-яким
    // рівнем доступу, в т. ч. private
    // статичний public-метод можна використовувати
    // в інших класах
    public static boolean isPrime(int n){
      boolean result = true;
      int max_div = (int) Math.sqrt(n);
      for (int p = 2; p <= max_div; p++){
        if (n % p == 0){
          result = false;
          break;
        }
      }
      return result;
    }
}
